comp110-worksheet-6/OxoBoard.cs

```diff
@@ -4,73 +4,97 @@
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+        private Mark[,] boardSize;
+        public OxoBoard( int width = 3, int height = 3, int inARow = 3 ) {
+            boardSize = new Mark[width, height];
+        }
 
-		// Return the contents of the specified square.
-		public Mark GetSquare(int x, int y)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // Return the contents of the specified square.
+        public Mark GetSquare(int x, int y) {
+            return boardSize[x, y];
+        }
 
-		// If the specified square is currently empty, fill it with mark and return true.
-		// If the square is not empty, leave it as-is and return False.
-		public bool SetSquare(int x, int y, Mark mark)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // If the specified square is currently empty, fill it with mark and return true.
+        // If the square is not empty, leave it as-is and return False.
+        public bool SetSquare(int x, int y, Mark mark) {
+            if (boardSize[x, y] == Mark.None) {
+                boardSize[x, y] = mark;
+                return true;
+            }
+            else {
+                return false;
+            }
+        }
 
-		// If there are still empty squares on the board, return false.
-		// If there are no empty squares, return true.
-		public bool IsBoardFull()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+        public bool IsBoardFull() {
+            for (int x = 0; x < boardSize.GetLength(0); x++) {
+                for (int y = 0; y < boardSize.GetLength(1); y++) {
+                    if (boardSize[x, y] == Mark.None) {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
-		public Mark GetWinner()
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner() {
+            // Check if Horizontal win
+            for (int horizontal = 0; horizontal < boardSize.GetLength(0); horizontal++) {
+                if (boardSize[horizontal, 0] == boardSize[horizontal, 1] && boardSize[horizontal, 1] == boardSize[horizontal, 2]) {
+                    return GetSquare(horizontal, 0);
```

Comment:
You are mixing `boardSize[]` and `GetSquare()` in this function -- it's best to be consistent with one or the other